apps/wallet-dashboard/components/Dialogs/migration/views/ConfirmMigrationView.tsx

```diff
@@ -0,0 +1,156 @@
+    Panel,
+    Title,
+    TitleSize,
+} from '@iota/apps-ui-kit';
+import { useGroupedMigrationObjectsByExpirationDate } from '@/hooks';
+import { Loader, Warning } from '@iota/ui-icons';
+import { Collapsible, useFormatCoin } from '@iota/core';
+import { IOTA_TYPE_ARG } from '@iota/iota-sdk/utils';
+import { summarizeMigratableObjectValues } from '@/lib/utils';
+import { MigrationObjectDetailsCard } from '@/components/migration/migration-object-details-card';
+import { DialogLayout, DialogLayoutBody, DialogLayoutFooter } from '../../layout';
+import { Transaction } from '@iota/iota-sdk/transactions';
+
+interface ConfirmMigrationViewProps {
+    basicOutputObjects: IotaObjectData[] | undefined;
+    nftOutputObjects: IotaObjectData[] | undefined;
+    onSuccess: () => void;
+    setOpen: (bool: boolean) => void;
+    isTimelocked: boolean;
+    migrateData:
+        | {
+              transaction: Transaction;
+              gasBudget: string | number | null;
+          }
+        | undefined;
+    isMigrationPending: boolean;
+    isMigrationError: boolean;
+    isSendingTransaction: boolean;
+}
+
+export function ConfirmMigrationView({
+    basicOutputObjects = [],
+    nftOutputObjects = [],
+    onSuccess,
+    setOpen,
+    isTimelocked,
+    migrateData,
+    isMigrationPending,
+    isMigrationError,
+    isSendingTransaction,
+}: ConfirmMigrationViewProps): JSX.Element {
+    const account = useCurrentAccount();
+
+    const {
+        data: resolvedObjects = [],
+        isLoading,
+        error: isGroupedMigrationError,
+    } = useGroupedMigrationObjectsByExpirationDate(
+        [...basicOutputObjects, ...nftOutputObjects],
+        isTimelocked,
+    );
+
+    const { totalNotOwnedStorageDepositReturnAmount } = summarizeMigratableObjectValues({
+        basicOutputs: basicOutputObjects,
+        nftOutputs: nftOutputObjects,
+        address: account?.address || '',
+    });
+
+    const [gasFee, gasFeeSymbol] = useFormatCoin(migrateData?.gasBudget, IOTA_TYPE_ARG);
+    const [totalStorageDepositReturnAmountFormatted, totalStorageDepositReturnAmountSymbol] =
+        useFormatCoin(totalNotOwnedStorageDepositReturnAmount.toString(), IOTA_TYPE_ARG);
+
+    return (
+        <DialogLayout>
+            <Header title="Confirmation" onClose={() => setOpen(false)} titleCentered />
+            <DialogLayoutBody>
+                <div className="flex h-full flex-col gap-y-md overflow-y-auto">
+                    {isGroupedMigrationError && !isLoading && (
+                        <InfoBox
+                            title="Error"
+                            supportingText="Failed to load migration objects"
+                            style={InfoBoxStyle.Elevated}
+                            type={InfoBoxType.Error}
+                            icon={<Warning />}
+                        />
+                    )}
+                    {isLoading ? (
+                        <LoadingIndicator text="Loading migration objects" />
```

Comment:
we could use skeleton instead of `LoadingIndicator` here, wdyt?

Reply:
Maybe its better to do it in another PR no?

Reply:
I'm working in this now :pray: